src/jsonrpc-virtual-machine.cpp

```diff
@@ -57,10 +57,25 @@ struct http_request_data {
     bool done;
 };
 
+// Performs additional client socket configuration
+static void setup_client_socket(struct mg_connection *c) {
+    // Minimize socket close time: by setting the linger time to 0.
+    // Avoid accumulating socket in TIME_WAIT state in rapid successive requests, which can consume all available ports.
+    // It is safe to do this because this client decides to close the connection, after all data is received.
+    struct linger so_linger {};
+    so_linger.l_onoff = 1;
+    so_linger.l_linger = 0; // do not wait
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+    auto socket = static_cast<MG_SOCKET_TYPE>(reinterpret_cast<size_t>(c->fd));
```

Comment:
A socket is an  regular signed ‘int’ on Unix and an unsigned `SOCKET` on Windows. I don’t know what mongoose does when it casts the damn thing to a pointer type. INVALID_SOCKET on windows is a large positive number. So converting to `size_t` is ok. But converting -1 to a pointer than to a `size_t` than back to `int` will probably overflow in an undefined way. So I dunno.